src/users/UserEdit.tsx

```diff
@@ -0,0 +1,22 @@
+import {
+  Edit,
+  SimpleForm,
+  TextInput,
+  EmailField,
+  ReferenceInput,
+  AutocompleteInput,
+} from "react-admin";
+
+export const UserEdit = () => (
+  <Edit resource="users">
+    <SimpleForm>
+      <TextInput source="username" label="Username" />
+      <TextInput source="first_name" label="First Name" />
+      <TextInput source="last_name" label="Last Name" />
+      <EmailField source="email" label="Email" />
```

Comment:
Why a field? Why not an input?

Btw there is no `EmailInput`, but you can use a `TextInput` with an `email()` validator.